### DISCUSSION TITLE
How to use a different runtime?

### DISCUSSION THREAD

**Discussion by boldandbusted** [rancher-sandbox/rancher-desktop#4073] [Q&A]
Howdy. Thanks for Rancher Desktop. If I wanted to run 'runsc' as part of gVisor in Rancher Desktop, how would I go about that? Or is it even possible? Cheers!

**Comment by adamkpickering** [comment]
Anything is possible if you have enough time and people power to put into it...

All I can give are suggestions. As you might be aware, RD runs docker, containerd etc inside a VM. So if you wanted to experiment with running gVisor in RD, you'd have to do so in that VM. You can get a shell inside the VM with `rdctl shell`. From there, you'd have to tweak the `k3s`, `dockerd` and maybe `containerd` config to use `runsc`. From there... I'm really not sure. My understanding of how `runsc` would fit might be completely wrong though!

Note that changes you make to the VM may be overwritten when you restart RD. Best of luck!
